src/main/java/io/jenkins/plugins/opentelemetry/queue/MonitoringQueueListener.java

```diff
@@ -39,53 +40,36 @@ public class MonitoringQueueListener extends QueueListener {
     public void postConstruct() {
         meter.gaugeBuilder(JenkinsSemanticMetrics.JENKINS_QUEUE_WAITING)
             .ofLongs()
-            .setDescription("Number of waiting items in queue")
+            .setDescription("Number of tasks in the queue with the status 'waiting', 'buildable' or 'pending'")
             .setUnit("1")
-            .buildWithCallback(valueObserver -> valueObserver.record(this.getUnblockedItemsSize()));
+            .buildWithCallback(valueObserver -> valueObserver.record((long)
+                Optional.ofNullable(Jenkins.getInstanceOrNull()).map(j -> j.getQueue()).
```

Comment:
NIT: `Optional.ofNullable(Jenkins.getInstanceOrNull()).map(j -> j.getQueue())` can be refactored to a a variable

Reply:
I'm not clear on how to convert this into a variable that would be refreshed on every invocation by the meter. Feel free to propose a code change.